### ISSUE TITLE
Pre-populate the 'entry' title from URI detection

### ISSUE THREAD

**Issue by sandysage** [mozilla-lockbox/lockbox-extension#314]
When creating a new online account, I want those credentials to be easily populated into my Lockbox as a new entry.

# Acceptance criteria
1. Entry title is set by URI
2. Entry title field is still editable

# Context
Research findings from Alpha testing indicates there is confusion around what an entry is and how it might be different from the website address. By pre-populating the entry name, as derived from URI, no additional interpretation is needed from the users. And by making the field editable, a user can still control their entry list as desired.

**Comment by jimporter** [comment]
Rather than doing this, why not just have the items in the list on the left display the first origin (we can only have one now, but eventually, we'll be able to have many) if there's no title? That way, nothing in the *data* needs to change, but the UI will be a little friendlier.

**Comment by sandysage** [comment]
If I understand correctly, that means:

1. user clicks +
2. the user has blank fields for entry name, website address, username, password, notes
3. if the user doesn't populate "entry name" then the list view displays the origin as the entry title

If I captured that correctly, then we're still faced with the risk that the user doesn't know what the entry field is for when creating that entry, and we're not working towards helping users to populate their Lockbox entries (alleviating some of the manual burden).

**Comment by jimporter** [comment]
I think it depends on where the "+" is. In the current management tab, you're essentially creating a new entry from scratch, so I don't think we need to pre-fill anything (or copy data from other fields); in fact, that would be pretty surprising to me as a user. However, if a user is clicking the "+" in the doorhanger, I think it makes sense to prefill the title *and* origin.

Still, I do think it would be nice to have the entry in the list show the title and fallback to the (first) origin if no title exists. That would cover the case of people creating a "raw" entry from the management tab, as well as explicitly deleting the entry title from a pre-filled entry from the doorhanger.

**Comment by sandysage** [comment]
Yup. I agree on all those points.

My take-away is that this feature entails:

- [ ] pre-fill title and origin from 'create' in the toolbar UI (the expected primary use case)
- [ ] no changes to the management tab for 'create'
- [ ] for the list view, show the title (as we have it today) and fallback to the first origin if no title exists
- [ ] evaluate the origin UI to make it harder to accidentally modify it (rather than making it not editable at all) - needs-ux

Does that generally cover it?

Since we're not expecting the management tab to be the primary UI for most users, I think this emphasizes that this work effort is dependent on #315.

**Comment by jimporter** [comment]
@sandysage Looks good to me!

**Comment by devinreams** [comment]
May be worth splitting into two issues.
